Unable to run `e build` with `Cache-Only` access status

<!--  As an open source project with a dedicated but small maintainer team, it can sometimes take a long time for issues to be addressed so please be patient and we will get back to you as soon as we can.
-->

I'm running into a reclient issue that's blocking my ability to run `e build`. I've verified that I have a valid auth token, and have attached that info below. Talking to @jkleinsc, we think this might be related to my permission's Cache-Only access level.

### Operating System
<!-- (Platform and Version) e.g. macOS 10.13.6 / Windows 10 (1803) / Ubuntu 18.04 x64 -->

### Expected Behavior
<!-- A clear and concise description of what you expected to happen. -->

### Actual Behavior
<!-- A clear and concise description of what actually happened. -->

### To Reproduce
<!--
Your best chance of getting this bug looked at quickly is to provide an example.
-->

### Screenshots
<!-- If applicable, add screenshots to help explain your problem. -->

### Additional Information
<!-- Add any other context about the problem here. -->

Error:
```
e build
Checking for build-tools updates
Running "git pull origin main --rebase --autostash" in /Users/anny/.electron_build_tools
From https://github.com/electron/build-tools
 * branch            main       -> FETCH_HEAD
build-tools is up-to-date
WARN We've made these temporary changes to your configuration:
 * removed deprecated goma property
Run "e sanitize-config" to make these changes permanent.
Downloading "https://dev-cdn.electronjs.org/reclient/credential-helper/v0.4.2/electron-rbe-credential-helper-darwin-arm64.tar.gz" into /Users/anny/.electron_build_tools/third_party/reclient.tar.gz
[==============================================================================================================================================================================================] 23.85MB/s 100% 0.0s
Running "autoninja -j 200 electron" in /Users/anny/code/opensource/electron/src/out/Testing
Proxy started successfully.
...
FAILED: obj/third_party/perfetto/src/trace_processor/sorter/sorter/trace_sorter.o
../../buildtools/reclient/rewrapper -cfg=../../buildtools/reclient_cfgs/chromium-browser-clang/rewrapper_mac.cfg -exec_root=/Users/anny/code/opensource/electron/src/ ../../third_party/llvm-build/Release+Asserts/bin/clang++ -MMD -MF obj/third_party/perfetto/src/trace_processor/sorter/sorter/trace_sorter.o.d -DDCHECK_ALWAYS_ON=1 -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D_FORTIFY_SOURCE=2 -D_LIBCPP_HARDENING_MODE=_LIBCPP_HARDENING_MODE_EXTENSIVE -D__ARM_NEON__=1 -DCR_XCODE_VERSION=1500 -DCR_CLANG_REVISION=\"llvmorg-20-init-1009-g7088a5ed-10\" -D_LIBCPP_DISABLE_VISIBILITY_ANNOTATIONS -D_LIBCXXABI_DISABLE_VISIBILITY_ANNOTATIONS -DCR_LIBCXX_REVISION=9c3fb06173cf5a291faf8823dfb82a3da6835bc3 -DTEMP_REBUILD_HACK -DNDEBUG -DNVALGRIND -DDYNAMIC_ANNOTATIONS_ENABLED=0 -DPERFETTO_IMPLEMENTATION -I../.. -Igen -I../../buildtools/third_party/libc++ -I../../third_party/perfetto -I../../third_party/perfetto/src/profiling/memory/include -I../../third_party/perfetto/include -Igen/third_party/perfetto/build_config -Igen/third_party/perfetto -Wall -Wextra -Wimplicit-fallthrough -Wextra-semi -Wunreachable-code-aggressive -Wthread-safety -Wunguarded-availability -Wno-missing-field-initializers -Wno-unused-parameter -Wno-psabi -Wloop-analysis -Wno-unneeded-internal-declaration -Wno-cast-function-type -Wno-deprecated-this-capture -Wno-invalid-offsetof -Wno-vla-extension -Wno-thread-safety-reference-return -Wshadow -Werror -fno-delete-null-pointer-checks -fno-ident -fno-strict-aliasing -fstack-protector -fcolor-diagnostics -fmerge-all-constants -fno-sized-deallocation -fcrash-diagnostics-dir=../../tools/clang/crashreports -mllvm -instcombine-lower-dbg-declare=0 -mllvm -split-threshold-for-reg-with-hint=0 -ffp-contract=off -fcomplete-member-pointers --target=arm64-apple-macos -mno-outline -Wno-builtin-macro-redefined -D__DATE__= -D__TIME__= -D__TIMESTAMP__= -ffile-compilation-dir=. -no-canonical-prefixes -ftrivial-auto-var-init=pattern -O2 -fno-math-errno -fno-omit-frame-pointer -gdwarf-4 -g1 -gdwarf-aranges -isysroot sdk/xcode_links/MacOSX14.0.sdk -mmacos-version-min=11.0 -fvisibility=hidden -Wheader-hygiene -Wstring-conversion -Wtautological-overlap-compare -Xclang -add-plugin -Xclang find-bad-constructs -Xclang -plugin-arg-find-bad-constructs -Xclang span-ctor-from-string-literal -Xclang -plugin-arg-find-bad-constructs -Xclang raw-ref-template-as-trivial-member -Xclang -plugin-arg-find-bad-constructs -Xclang check-stack-allocated -Xclang -plugin-arg-find-bad-constructs -Xclang check-allow-auto-typedefs-better-nested -Xclang -add-plugin -Xclang raw-ptr-plugin -Xclang -plugin-arg-raw-ptr-plugin -Xclang check-raw-ptr-to-stack-allocated -Xclang -plugin-arg-raw-ptr-plugin -Xclang disable-check-raw-ptr-to-stack-allocated-error -Xclang -plugin-arg-raw-ptr-plugin -Xclang raw-ptr-exclude-path=../../third_party/dawn/ -Xclang -plugin-arg-raw-ptr-plugin -Xclang check-raw-ptr-fields -Xclang -plugin-arg-raw-ptr-plugin -Xclang check-raw-ref-fields -Wenum-compare-conditional -Wno-c++11-narrowing-const-reference -Wno-missing-template-arg-list-after-template-kw -Wno-dangling-assignment-gsl -std=c++20 -Wno-trigraphs -fno-exceptions -fno-rtti -nostdinc++ -isystem../../third_party/libc++/src/include -isystem../../third_party/libc++abi/src/include -fvisibility-inlines-hidden -c ../../third_party/perfetto/src/trace_processor/sorter/trace_sorter.cc -o obj/third_party/perfetto/src/trace_processor/sorter/sorter/trace_sorter.o
reclient[79102439-2831-49af-b348-e6c8433197b0]: LocalErrorResultStatus: this build has encountered too many local fallbacks. This means that the ratio of actions that failed remotely is equal to or above the preconfigured threshold of 50.0%
ninja: build stopped: subcommand failed.
RBE Stats: down 319.67 KB, up 0 B, 24 cache hits, 408 racing locals, 200 local failures
ERROR Error: Command failed: autoninja -j 200 electron
    at checkExecSyncError (node:child_process:885:11)
    at Object.execFileSync (node:child_process:921:15)
    at Object.depotExecFileSync [as execFileSync] (/Users/anny/.electron_build_tools/src/utils/depot-tools.js:149:23)
    at runNinja (/Users/anny/.electron_build_tools/src/e-build.js:86:9)
    at Command.<anonymous> (/Users/anny/.electron_build_tools/src/e-build.js:140:9)
    at Command.listener [as _actionHandler] (/Users/anny/.electron_build_tools/node_modules/commander/lib/command.js:480:17)
    at /Users/anny/.electron_build_tools/node_modules/commander/lib/command.js:1234:65
    at Command._chainOrCall (/Users/anny/.electron_build_tools/node_modules/commander/lib/command.js:1151:12)
    at Command._parseCommand (/Users/anny/.electron_build_tools/node_modules/commander/lib/command.js:1234:27)
```

Auth details:
```
Authentication Status: Authenticated
Since:     2024-08-26 15:46:33 -0400 EDT
Expires:   2024-08-28 15:46:33 -0400 EDT
Cluster:   cas.notgoma.com
Principal: github/yangannyx
Access:    Cache Only
```

--- a/src/utils/reclient.js
+++ b/src/utils/reclient.js
@@ -14,7 +14,7 @@ const reclientHelperPath = path.resolve(
 );
 const rbeServiceAddress = 'rbe.notgoma.com:443';
 
-const CREDENTIAL_HELPER_TAG = 'v0.4.2';
+const CREDENTIAL_HELPER_TAG = 'v0.4.3';
 
 function downloadAndPrepareReclient(config, force = false) {
   if (config.reclient === 'none' && !force) return;
@@ -99,11 +99,20 @@ function reclientEnv(config) {
     return {};
   }
 
-  return {
+  let reclientEnv = {
     RBE_service: config.reclientServiceAddress || rbeServiceAddress,
     RBE_experimental_credentials_helper: getHelperPath(config),
     RBE_experimental_credentials_helper_args: 'print',
   };
+
+  const result = childProcess.spawnSync(reclientHelperPath, ['flags'], {
+    stdio: 'pipe',
+  });
+  if (result.status === 0) {
+    const extraArgs = JSON.parse(result.stdout.toString());
+    reclientEnv = Object.assign(reclientEnv, extraArgs);
+  }
+  return reclientEnv;
 }
 
 function ensureHelperAuth(config) {

